### ISSUE TITLE
ActivityPub Client Puppet(?)

### ISSUE THREAD

**Issue by shiribailem** [friendica/friendica#13392]
This idea came up looking into things like the video questions and how the response is largely "other platforms do it better and none of us are comfortable with handling that", and I thought of a middle ground. Using a separate feature request because this is a bit more versatile than just a single function.

The basic idea I'm thinking is that you might be able to do things like connect a Peertube account to your normal Friendica account. I use the term puppet for lack of a better term (maybe bridge? but that's confusing as well).

My general thought is if you could connect your Friendica account to other AP accounts for specific content. I might for instance have an option under "More" to open a post in Peertube so I can reply easily with a video.

Or maybe since Lemmy misbehaves some, I might open a Lemmy post with my separate Lemmy account.

The basic form of this would just be to forward me to that post but instead of on the original site, accessing it from a different site I have an account with. I suspect this might have to use a client connection to get the link?

**Comment by MrPetovan** [comment]
As far as I know, this would require some form of C2S (Client To Server) capabilities on the target platforms. The current main way ActivityPub is used is between servers, or S2S (Server To Server). This implementation is required to enable federation. However a C2S implementation isn't required, and is rarely done that I know of. Mastodon set the de facto standards for client API, and it isn't an ActivityPub C2S implementation, which makes any potential C2S implementer an outlier without access to all the existing Mastodon API-compatible apps and clients.

Unless you just had in mind "redirect to remote profile compose page" for your "Open a post in X" menu item?

**Comment by shiribailem** [comment]
Little of A, little of B.

Hypothetical example involving friendica, AP-site, and a peertube site:

From friendica, viewing a post from an AP-site, I could click a button and have it open the ap-site post from the peertube site, allowing me to reply to a thread easily with a video rather than needing to copy the post url, open the peertube site, and then search that url to pull it up.

But I get if that's not feasible, I just assumed their might be a way from the client api to search a remote post and return a local link address to that post.

**Comment by MrPetovan** [comment]
We do have this feature in Friendica, and I know for a fact Mastodon does too, but I don't know if Peertube has this feature.

**Comment by annando** [comment]
I don't think that Peertube is made in a way that you can use it as a regular Fediverse client where you can subscribe to users and can read and comment their posts.
